From 553f9c8706dba454ae06f9ebf303e79401926051 Mon Sep 17 00:00:00 2001
From: Yoni Izutskiver <yizutskiver@habana.ai>
Date: Sun, 09 Jan 2022 11:16:29 +0200
Subject: [PATCH] [SW-69272] gaudi2: pci: fix pba register access

The HW reset value for MSIx PBA offset register is incorrect, and should
be overwritten before the PCIe link is enabled. Originally the overwrite
was done at a point where perst_n can still be asserted, and DBI space
is not available for writing. Moved the register overwrite to a stage
where perst_n is de-asserted.

Change-Id: I25cb170b150f922c827498287f97a6995306f3b0
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index f8ced3b..e91149a 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -124,8 +124,6 @@
 	/* set reset source */
 	hl_reset_src_init();
 
-	/* This W/A is to be done before pcie link is established */
-	set_pci_msix_pba_offset_val();
 }
 
 void hl_plat_init(void)
diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index 3ed456c..d42e07a 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -137,9 +137,6 @@
 
 	gaudi2_axi_drain_ena();
 
-	/* This W/A is to be done before pcie link is established */
-	set_pci_msix_pba_offset_val();
-
 	/* set reset source */
 	hl_reset_src_init();
 }
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 052378a..e99fca7 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -552,6 +552,9 @@
 	/* PCIe disable LTR */
 	preboot_disable_pcie_ltr();
 
+	/* set MSIx PBA offset */
+	set_pci_msix_pba_offset_val();
+
 	/* Enable and Mask LTSSM (erratum 102) */
 	regmap_writel(3, mmPCIE_AUX_LTSSM_EN);
 }
